### ISSUE TITLE
[Crash] Custom tabs crash when the system language is Romanian

### ISSUE THREAD

**Issue by sv-ohorvath** [mozilla-mobile/fenix#9392]
> Sentry https://sentry.prod.mozaws.net/operations/fenix-beta/issues/7637967/

## Steps to reproduce
1. Set device language to Romanian. (cannot reproduce with Spanish, German, Italian or French)
2. Open a custom tab: from Gmail, or the account login page, or any support link from the About page.
### Expected behavior
Custom tabs open.
### Actual behavior
App crashes.
[logcat.txt](https://github.com/mozilla-mobile/fenix/files/4380308/crash.in.custom.tabs.txt)

### Device information

* Android device: Nexus 9 (Android 7.1.1), Pixel 4 (Android 10)
* Fenix version: Beta 4.2.0, Nightly 3/24. Is reproducible since Beta 4.1.0

`    --------- beginning of crash
2020-03-25 12:27:26.824 22946-22946/? E/AndroidRuntime: FATAL EXCEPTION: main
    Process: org.mozilla.fenix.beta, PID: 22946
    java.lang.RuntimeException: Unable to start activity ComponentInfo{org.mozilla.fenix.beta/org.mozilla.fenix.customtabs.ExternalAppBrowserActivity}: java.util.UnknownFormatConversionException: Conversion = '&'
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2665)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2726)
        at android.app.ActivityThread.-wrap12(ActivityThread.java)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1477)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:154)
        at android.app.ActivityThread.main(ActivityThread.java:6119)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776)
     Caused by: java.util.UnknownFormatConversionException: Conversion = '&'
        at java.util.Formatter$FormatSpecifier.conversion(Formatter.java:2730)
        at java.util.Formatter$FormatSpecifier.<init>(Formatter.java:2759)
        at java.util.Formatter$FormatSpecifierParser.<init>(Formatter.java:2591)
        at java.util.Formatter.parse(Formatter.java:2524)
        at java.util.Formatter.format(Formatter.java:2472)
        at java.util.Formatter.format(Formatter.java:2426)
        at java.lang.String.format(String.java:2670)
        at android.content.res.Resources.getString(Resources.java:406)
        at android.content.Context.getString(Context.java:492)
        at org.mozilla.fenix.customtabs.CustomTabToolbarMenu.<init>(CustomTabToolbarMenu.kt:21)
        at org.mozilla.fenix.components.toolbar.BrowserToolbarView.<init>(BrowserToolbarView.kt:72)
        at org.mozilla.fenix.browser.BaseBrowserFragment.initializeUI(BaseBrowserFragment.kt:57)
        at org.mozilla.fenix.customtabs.ExternalAppBrowserFragment.initializeUI(ExternalAppBrowserFragment.kt:1)
        at org.mozilla.fenix.browser.BaseBrowserFragment.onViewCreated(BaseBrowserFragment.kt:1)
        at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:94)
        at androidx.fragment.app.FragmentManager.addAddedFragments(FragmentManager.java:5)
        at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:54)
        at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:11)
        at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:4)
        at androidx.fragment.app.FragmentManager.dispatchStateChange(FragmentManager.java:5)
        at androidx.fragment.app.Fragment.performActivityCreated(Fragment.java:9)
        at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:99)
        at androidx.fragment.app.FragmentManager.moveFragmentToExpectedState(FragmentManager.java:4)
        at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:218)
        at androidx.fragment.app.FragmentManager.dispatchStateChange(FragmentManager.java:3)
        at androidx.fragment.app.FragmentActivity.onStart(FragmentActivity.java:9)
        at androidx.appcompat.app.AppCompatActivity.onStart(AppCompatActivity.java:1)
        at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1248)
        at android.app.Activity.performStart(Activity.java:6696)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2628)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2726) 
        at android.app.ActivityThread.-wrap12(ActivityThread.java) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1477) 
        at android.os.Handler.dispatchMessage(Handler.java:102) 
        at android.os.Looper.loop(Looper.java:154) 
        at android.app.ActivityThread.main(ActivityThread.java:6119) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:886) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:776) `

**Comment by st3fan** [comment]
This is definitely looking like a locale specific crash that may happen for other locales too.

**Comment by st3fan** [comment]
@flodolo we're trying to understand why Romanian has an isssue with a `Conversion = '&'` ...

**Comment by pocmo** [comment]
It looks this String may be the culprit. The placeholder is definitely wrong. I'll verify if that causes the crash in a minute:
https://github.com/mozilla-mobile/fenix/blob/master/app/src/main/res/values-ro/strings.xml#L99

**Comment by pocmo** [comment]
Just confirmed that this string translation is causing the crash.

```XML
<string name="browser_menu_powered_by">CU TEHNOLOGIE %1&amp;s</string>
```

should  be

```XML
<string name="browser_menu_powered_by">CU TEHNOLOGIE %1$s</string>
```

**Comment by flodolo** [comment]
[Fixed in Pontoon](https://pontoon.mozilla.org/ro/android-l10n/mozilla-mobile/fenix/app/src/main/res/values/strings.xml/?search=browser_menu_powered_by&string=197450), should be in the Import PR tomorrow (it will be in the repo in a few minutes, but need to poke RelEng to force an export before midnight UTC).

**Comment by liuche** [comment]
We've gotten a new L10N import 3 days ago, so let's re-check this.

**Comment by pocmo** [comment]
@Pike Just to make you aware of this. I am not sure if we could prevent that in `compare-locales`? It seems like Android is trying to parse the placeholder formatting rules which may not be easy to emulate.

**Comment by Pike** [comment]
I think this is a variant of https://bugzilla.mozilla.org/show_bug.cgi?id=1621968

**Comment by sflorean** [comment]
Verified as fixed on Nightly build 200331.
Devices: Samsung Galaxy Note10 (Android 10), Nokia 6 (Android 7.1.1), Xiaomi Mi Pad 2 (Android 5.1).

**Comment by ebalazs-sv** [comment]
Verified as fixed on Beta Migration 75.0.0-beta.3 4/1 with Huawei P9 Lite (Android 7) and HTC Desire 530 (Android 6.0.1).
